content/en/docs/reference/labels-annotations-taints/_index.md

```diff
@@ -493,8 +493,15 @@ Example: `kubernetes.io/enforce-mountable-secrets: "true"`
 Used on: ServiceAccount
 
 The value for this annotation must be **true** to take effect.
-This annotation indicates that Pods running as this ServiceAccount may only reference
-Secret API objects specified in the ServiceAccount's `secrets` field.
+When set to **true**, this annotation enforces the following rules for Pods running as this ServiceAccount:
+
+1. Secrets mounted as volumes must be listed in the ServiceAccount's `secrets` field.
+2. Secrets referenced in `envFrom` for containers or `initContainers` must also be listed in the ServiceAccount's `secrets` field.
+3. Secrets referenced in a Pod's `imagePullSecrets` must be present in the ServiceAccount's `imagePullSecrets` field.
+
+While the annotation's name suggests it only concerns the mounting of secrets,
+its enforcement also extends to other ways Secrets are used in the context of a Pod.
+Therefore, it is crucial to ensure that all the referenced secrets are correctly specified in the ServiceAccount.
```

Comment:
```suggestion
Therefore, it is crucial to ensure that all the referenced Secrets are correctly specified in the ServiceAccount.
```

Reply:
I would put this advice in the ServiceAccount page, not here. And strongly encourage readers to go look there.